File: 151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let res = ""
    s = s.trim().split(" ").reverse()
    for(let word of s){
        res+= word === ""? "": word + " "
    }
    return res.trim()
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = [c for c in s.strip().split(" ") if c != ""]
        start, end = 0, len(s) - 1
        while start < end:
            s[start], s[end] = s[end], s[start]
            start+=1; end-=1
        return " ".join(s)
        